flood_ml/data_utils.py

```diff
@@ -0,0 +1,64 @@
+"""Data functions for Flood CNN model."""
+
+from constants import GEO_FEATURES, MAP_HEIGHT, MAP_WIDTH, N_FLOOD_MAPS
+
+import numpy as np
+from numpy.lib.stride_tricks import sliding_window_view
+import tensorflow as tf
+
+
+def temporal_window_view(data, window):
+    """Creates a sliding window view over 1D temporal data.
+
+    Args:
+        data: A 2D tensor, where the first axis is the batch dimension and the
+            second axis is the temporal data. Shape: [B, T].
+        window: The window size.
+
+    Returns:
+        A 3D tensor of shape [B, T, window]. In order to maintain the same T,
+        we use zero padding at the beginning of the temporal data.
+    """
+    B = data.shape[0]
+    padding = np.zeros([B, window - 1])
+    data = np.concatenate([padding, data], axis=-1)
+    return sliding_window_view(data, window, axis=-1)
+
+
+def fake_input_batch(
+    batch_size,
+    height=None,
+    width=None,
+    rainfall_duration=24,
+    flood_maps=None,
+    include_flood_maps=False,
+):
```

Comment:
Let's add a type annotations for the return type and some information about what we return to the docstring.

Reply:
Sorry, I missed the return statement in the docstring. Do you want type annotations for the code in general, or was this because the docstring was also missing the information?

Reply:
I'm into type annotations generally. I do think they're useful for documentation & readability. 

We've set up MyPy to do type checking for the other packages, but it needs type annotations to do anything useful. MyPy just ignores functions when they don't have type annotations

```python
def func():
  return None

1 + func()
```
MyPy Type checking will be perfectly happy

```python
def func() -> None:
  return None

1 + func()
```
now MyPy will actually diagnose a type error

Reply:
Sure, I'm generally a fan of annotations too :) I think with ML code it can become a bit unwieldy. I'm not sure if specifying a generic type like `tf.Tensor` is actually that useful. We could define more detailed annotations, for example using [jaxtyping](https://github.com/patrick-kidger/jaxtyping), to include information about expected shapes, which I'm currently including in docstrings instead. My main question here is whether this would be useful to TNS and whether they would want to maintain the convention post-fellowship, since these are non-standard annotations and IMO not the most intuitive.

Maybe for more general code like this function, we can include type annotations. And within the model itself, I'll see if it still makes sense.

Reply:
Yep, I think that makes sense. Happy to leave them off where they aren't helpful.